drivers/pcie/host/pcie.c

```diff
@@ -370,6 +370,93 @@ pcie_bdf_t pcie_bdf_lookup(pcie_id_t id)
 	return PCIE_BDF_NONE;
 }
 
+/* Forward declaration needed since scanning a device may reveal a bridge */
+static bool scan_bus(int bus, const struct pcie_scan_opt *opt);
+
+static bool scan_dev(int bus, int dev, const struct pcie_scan_opt *opt)
```

Comment:
it's probably really unlikely to happen, but shouldn't you have a call counter here to limit the recursive calls to something "sane" (whatever that could be) ?

Reply:
You've essentially got some weird or broken HW in that case. Not sure if it's worth it, but a call counter sounds a bit arbitrary. Another option would be to have a bitmask of already processed bus numbers, but that might also be excessive overhead considering that this really indicates that there's something wrong with the HW.

Reply:
I thought a bit more about this, and I don't think we should bother. In the Linux case it always allocates and programs new bus numbers to found busses (by writing a new number to the register), so it doesn't even try to do some kind of circular dependency kind of detection.